### ISSUE TITLE
Virus scan exception is not working

### ISSUE THREAD

**Issue by thirupathicys** [e2guardian/e2guardian#577]
e2guardian 5.3.3 & 5.3.4
I am trying to exclude certain extensions from being scanned by clamav. I added .txt, .php, .html, ..etc to exceptionvirusextensionlist file and then the list has been enabled in e2guardianf1.conf but it seems the files are being scanned by clamav even though it is in exception list.
Can you plz guide me in the right direction to skip files from being scanned?

**Comment by fredbcode** [comment]
How did you see this that ? 
Your storyboard is up to date ?

https://github.com/e2guardian/e2guardian/blob/v5.3/configs/common.story#L208

**Comment by thirupathicys** [comment]
Yes , storyboard is up to date. 
Tested by loading eicar txt file and e2guardian is blocking that txt file.

**Comment by thirupathicys** [comment]
Developers plz help to sort this issue?

**Comment by philipianpearce** [comment]
Hi, 

Can you please set storyboardtrace = on in e2guardian.conf and then post the output from the syslog. Only call the one url to test while in this mode as it gives verbose output. 

Also which version of e2g are you using (e2guardian -v will show this). 

Philip 

----- Original Message -----

**Comment by thirupathicys** [comment]
Hi ,
After enabling storyboardtrace =on , syslog output pasted below. For testing i have hosted the wesever and load txt file with clam signatre. After that, I enabled to skip .txt extension file for scanning.But it is scanned and blocked by e2g. Below is the syslog output,

 "Feb 10 14:43:54 itma2 clamd[31839]: Mon Feb 10 14:43:54 2020 -> ~/tmp/cswiUGi6: Clamav.Test.File-7(44d88612fea8a8f36de82e1278abb02f:68) FOUND"


e2g version :  output
e2guardian 5.3.3

Built with:  '--prefix=/usr' '--enable-clamd=yes' '--with-proxyuser=e2guardian' '--with-proxygroup=e2guardian' '--sysconfdir=/etc' '--localstatedir=/var' '--enable-icap=yes' '--enable-commandline=yes' '--enable-email=yes' '--enable-ntlm=yes' '--mandir=${prefix}/share/man' '--infodir=${prefix}/share/info' '--enable-pcre=yes' 'CPPFLAGS=-mno-sse2 -g -O2'

access.log output:

1581327054.850		-	192.168.2.125	192.168.2.125	http://10.20.30.130/test.txt	GET	403	68	-	-	-	24	-	0	*INFECTED* * *DENIED* Virus or bad content detected. Clamav.Test.File-7	0	Content scanning	Default	1

Please help me to solve the issue.
Thanks
Thiru

**Comment by fredbcode** [comment]
There is nothing more in syslog ?

**Comment by fredbcode** [comment]
@thirupathicys which OS are you using ?

**Comment by fredbcode** [comment]
@philipianpearce  I quickly checked 

Maybe I missed something but I guess there is something weird here 
https://github.com/e2guardian/e2guardian/blob/v5.3/src/ConnectionHandler.cpp#L4370

check_content is outside if(!checkme.noviruscheck) ?

**Comment by thirupathicys** [comment]
Hi,
I am using "Debian 10" operating system.

**Comment by thirupathicys** [comment]
> @thirupathicys which OS are you using ?

**Comment by fredbcode** [comment]
Please try https://gitlab.com/fredbcode/e2guardian/-/tree/packages/bug577

**Comment by philipianpearce** [comment]
@thirupathicys - with storyboardtrace = on you should have a lot of lines with SB: in the syslog with an e2guardian label - I need these in order to help diagnose where the issue is. 

----- Original Message -----

**Comment by thirupathicys** [comment]
> Please try https://gitlab.com/fredbcode/e2guardian/-/tree/packages/bug577

Hi fred, thank you very much for the quick response. I tried your deb file and the files are still being scanned. I have a webserver running locally with just eicar.txt file and e2guardian is not skipping the ".txt" extension from virus scan. I have enabled the virus scan exemption in e2guardianf1.conf and added ".txt" in exceptionvirusextensionlist under contentscanners in lists folder.



> @thirupathicys - with storyboardtrace = on you should have a lot of lines with SB: in the syslog with an e2guardian label - I need these in order to help diagnose where the issue is.

Hi philip, attaching the lines with storyboardtrace on.
[storyboardtrace.txt](https://github.com/e2guardian/e2guardian/files/4203198/storyboardtrace.txt)

Thanks!
